### ISSUE TITLE
ODE Form C readout: negative [scaling] predictions silently clamped to zero

### ISSUE THREAD

**Issue by roninsightrx** [FeRx-NLME/ferx-core#1020]
## Summary

`ode_predictions` clamps the **final, post-readout** prediction vector at zero. That is right for a concentration, but it is applied unconditionally — including to a Form C `[scaling]` `y[CMT=N] = <expression>` readout, which is an arbitrary user expression and is often legitimately negative.

Any negative part of such a readout silently becomes `0`. No warning, no error; the fit converges and returns plausible-looking parameters.

## Reprex

`negpred.ferx` — the readout is a straight line crossing zero:

```
model negpred
[parameters]
  theta TVA(1.0, 0.1, 10.0)
  omega ETA_A ~ 0.01
  sigma ADD_ERR ~ 1.0 (sd)
[individual_parameters]
  A = TVA * exp(ETA_A)
[structural_model]
  ode(states=[clock])
[odes]
  d/dt(clock) = 1
[scaling]
  y[CMT=1] = A * (clock - 5)
[error_model]
  DV ~ additive(ADD_ERR)
[fit_options]
  method = focei
  gradient = fd
  covariance = false
```

`negpred.csv`:

```
ID,TIME,DV,EVID,AMT,CMT,MDV
1,0,0,1,0,1,1
1,2,-3,0,0,1,0
1,4,-1,0,0,1,0
1,6,1,0,0,1,0
1,10,5,0,0,1,0
```

```r
library(ferx)
ferx_predict("negpred.ferx", "negpred.csv")
#>   ID TIME PRED   expected
#> 1  1    2    0         -3
#> 2  1    4    0         -1
#> 3  1    6    1          1
#> 4  1   10    5          5
```

At `TVA = 1` the readout is `clock - 5`. The two negative points are returned as `0`; the positive ones are exact.

## Cause

`src/ode/predictions.rs`:

```rust
    // Clamp negative predictions to zero (ODE solver overshoot guard).
    // NaN intentionally NOT clamped ...
    clamp_negative_predictions(&mut predictions);      // L2501
```

```rust
fn clamp_negative_predictions(predictions: &mut [f64]) {   // L1592
    for p in predictions.iter_mut() {
        if *p < 0.0 { *p = 0.0; }
    }
}
```

This runs on the final vector, i.e. **after** `OdeReadout::PerCmt` / `Single` has evaluated the `[scaling]` expression (`L1534`). The overshoot guard is a statement about a *state* (an amount or concentration cannot go below zero), but it is being enforced on the *readout*, which the code itself documents as arbitrary:

```rust
    /// Both Form C variants (`Single` and `PerCmt`) call arbitrary ...   // L1818
```

## Why it matters

Any signed readout is affected: change from baseline, a z-score, a difference from a comparator, an effect on a signed scale — and logits.

Concretely, it silently breaks the standard model-based meta-analysis of a bounded endpoint. The published construction (Boucher & Bennetts; Bracis et al., *CPT:PSP* 2026) transforms a responder proportion as

```
DV       = sqrt(N) * logit(observed proportion)
y[CMT=1] = sqrt(N) * logit(predicted proportion)
```

which is negative for every arm with a response rate below 50% — most of a real dataset. Fitting the published canakinumab/RA ACR20 model against its published Monolix estimates:

| | ferx | Monolix (published) |
|---|---|---|
| T50 (weeks) | 2.67 | 4.08 |
| BSV ω (SD) | 0.080 | 0.253 |
| residual σ (SD) | 3.03 | 2.47 |
| log-OR canakinumab | 1.111 | 1.104 |

Every placebo-arm prediction comes back as exactly `0`. σ inflates to absorb the mismatch and ω collapses. The drug effects survive because they are identified largely by the arms that do exceed 50%.

The companion continuous-endpoint model (WOMAC pain / SE, strictly positive) reproduces the same published analysis to three digits, which localises it to the sign of the readout rather than anything else in the model.

## Suggested fix

Apply the clamp only where the non-negativity is a physical property of the quantity — a bare state/concentration readout (`obs_scale`, analytical PK, generated `ode_template`) — and leave Form C `[scaling]` expressions unclamped. The NaN behaviour documented in the same comment should be kept either way.

If the clamp is load-bearing for the "missing `PerCmt` entry surfaces loudly" case mentioned in the comment, that check could be made explicit rather than riding on the sign.

A per-model opt-out (`[fit_options] clamp_predictions = false`) would be a workable interim, but the default is the problem: the current behaviour is silent, and it is not detectable from the fit output.

## Environment

```
ferx-r     0.2.0.9000
ferx-core  0.2.0
R          4.5.3, aarch64-apple-darwin
```

**Comment by roninsightrx** [comment]
Collected into the MBMA umbrella issue #1032, which has context on why this class of bug shows up in meta-analysis models specifically.
